analyze.go

```diff
@@ -108,19 +108,17 @@ func analyze(fname string, stmt *Directive, term string, ctx blockCtx, options *
 		}
 	}
 
-	// If DirectiveSources was not provided, try to find the directive in legacy way.
-	// We want to use DirectiveSources to indicate the OSS/N+ version, and dynamic
-	// modules we want to include for validation. However, invokers used MatchFuncs
-	// and a directive map before. This is a transition plan. After MatchFuncs is deleted
-	// from ParseOptions, this if block should be deleted.
+	// If DirectiveSources was not provided, use the default map
 	if len(options.DirectiveSources) == 0 {
 		masks, knownDirective = directives[directiveName]
-		if !knownDirective {
-			for _, matchFn := range options.MatchFuncs {
-				if masks, knownDirective = matchFn(stmt.Directive); knownDirective {
-					break
-				}
-			}
+	}
+
+	if knownDirective {
+		// Check if the directive has human-defined special definition
+		// If so, use it to do validation
```

Comment:
I found the comment to be a bit oddly worded, but I think you shouldn't need it anyway. I think if `forcedMap` were named better then the code could more clearly communicate what is happening here.